src/Expanders.jl

```diff
@@ -447,27 +447,45 @@ end
 # --------
 
 function Selectors.runner(::Type{ExampleBlocks}, x, page, doc)
-    matched = match(r"^@example[ ]?(.*)$", x.language)
-    matched === nothing && error("invalid '@example' syntax: $(x.language)")
     # The sandboxed module -- either a new one or a cached one from this page.
-    name = matched[1]
+    name = match(r"^@example[ ]?(.*)$", split(x.language, ';', limit = 2)[1])[1]
     sym  = isempty(name) ? gensym("ex-") : Symbol("ex-", name)
     mod  = get!(page.globals.meta, sym, Module(sym))::Module
+
+    # "parse" keyword arguments to example (we only need to look for execute = false)
+    execute = !occursin(r"execute\s*=\s*false", x.language)
+
     # Evaluate the code block. We redirect stdout/stderr to `buffer`.
     result, buffer = nothing, IOBuffer()
-    for (ex, str) in Utilities.parseblock(x.code, doc, page)
-        (value, success, backtrace, text) = Utilities.withoutput() do
-            cd(dirname(page.build)) do
-                eval(mod, Expr(:(=), :ans, ex))
+    if execute
+        # check if there is any code wating
+        if isdefined(mod, :__WAITING__CODE__)
```

Comment:
I guess the "waiting code" could also be stored directly in e.g. `page.globals.meta`, so that we wouldn't create arbitrary stuff in the module, but either way is fine.

If we do go with a variable in the module, I think it would be good to namespace it even more (e.g. `__DOCUMENTER__WAITING__CODE__`). Would make any potential conflict even less likely and it would make it clear where this variable comes from, e.g. when someone does `names(@__MODULE__)` in an at-example. I would also mention it in the manual that Documenter sets that variable.

Reply:
> I guess the "waiting code" could also be stored directly in e.g. page.globals.meta, so that we wouldn't create arbitrary stuff in the module, but either way is fine.

Yes, that seems much cleaner.